packages/@headlessui-react/src/components/popover/popover.tsx

```diff
@@ -948,8 +948,8 @@ function PanelFn<TTag extends ElementType = typeof DEFAULT_PANEL_TAG>(
         ? (event: ReactFocusEvent) => {
             let el = event.relatedTarget as HTMLElement
             if (!el) return
-            if (!internalPanelRef.current) return
-            if (internalPanelRef.current?.contains(el)) return
```

Comment:
It's weird that this event handler uses state whereas the `handleKeyDown` one does not. Likely because useEvent is being used?

Because this is an event handler I feel like `internalPanelRef.current` should still be used.